lib/pure/os.nim

```diff
@@ -3217,6 +3184,57 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
+proc sameFileContent*(path1, path2: string; checkSize = false; checkFiles = false; bufferSize = 8192.Positive): bool {.
```

Comment:
maybe leave as unresolved TODO for future PR:

since this PR exposes `bufferSize`, maybe it's the time to ask this:
I'm fine with exposing `bufferSize`, but can't we use the OS-specific optimal `bufferSize` when user doesn't explicitly set a value, instead of hardcoding it to `8192` ?
(yes, on some systems, it'd be 8192, but not all, and you get more portable/optimial code if user doesnt' need to specify it).

this could be:
```nim
proc getDefaultBufferSize*(): int = ...
proc sameFileContent*(path1, path2: string; checkSize = false; checkFiles = false; bufferSize = getDefaultBufferSize()): bool ...
```

Reply:
On modern times I dont know if theres such thing, for example you can add gigabytes of ram to a virtualbox with a click. :shrug:

Reply:
https://github.com/nim-lang/RFCs/issues/286

Reply:
=> https://github.com/nim-lang/Nim/pull/16023